compose-table/src/main/java/org/dhis2/composetable/ui/DataSetTableUi.kt

```diff
@@ -595,222 +596,225 @@ fun DataTable(
     tableList: List<TableModel>,
     editable: Boolean = true,
     tableColors: TableColors? = null,
+    tableDimensions: TableDimensions = TableTheme.dimensions,
     tableSelection: TableSelection = TableSelection.Unselected(),
     inputIsOpen: Boolean = false,
     tableInteractions: TableInteractions = object : TableInteractions {}
 ) {
-    if (!editable && !tableList.all { it.areAllValuesEmpty() }) {
-        TableItem(
-            tableModel = tableList.first(),
-            tableColors = tableColors,
-            tableInteractions = tableInteractions
-        )
-    } else if (editable) {
-        TableList(
-            tableList = tableList,
-            tableColors = tableColors,
-            tableSelection = tableSelection,
-            inputIsOpen = inputIsOpen,
-            tableInteractions = tableInteractions
-        )
+    val localDensity = LocalDensity.current
+    var tableTotalWidth by remember {
+        mutableStateOf(0.dp)
+    }
+
+    val onSizeChanged: (IntSize) -> Unit = {
+        tableTotalWidth = with(localDensity) {
+            it.width.toDp()
+        }
+    }
+
+    TableTheme(tableColors, tableDimensions.copy(totalWidth = tableTotalWidth)) {
```

Comment:
This is ok for now as it was implemented like this but I think we should remove TableTheme from the Composable in the near future.